controller/api/destination/watcher/cluster_store.go

```diff
@@ -221,8 +221,10 @@ func (cs *ClusterStore) addCluster(clusterName string, secret *v1.Secret) error
 		stopCh,
 	}
 
-	remoteAPI.Sync(stopCh)
-	metadataAPI.Sync(stopCh)
+	go func() {
```

Comment:
we could even go a step further and sync each of these in their own goroutine to sync them in parallel.